dev-requirements.in

```diff
@@ -13,3 +13,5 @@ google-cloud-bigquery
 google-cloud-bigquery-storage
 IPython
 torch
+tensorflow<=2.8.1
```

Comment:
Why does tensorflow need an upper limit?

Reply:
@samhita-alla It is because of the following [issue](https://github.com/flyteorg/flytekit/pull/1243#issuecomment-1287650838) that I was getting locally, so as a safety measure I added an upper limit.

Reply:
For now, I've decided to remove the upper bound to prevent merge conflict